components/tag-list/tag-list.js

```diff
@@ -222,15 +267,55 @@ class TagList extends LocalizeCoreElement(ArrowKeysMixin(LitElement)) {
 		return items.filter(({ isHidden }) => !isHidden);
 	}
 
-	_getTagListItems() {
+	async _getTagListItems() {
 		const slot = this.shadowRoot && this.shadowRoot.querySelector('slot');
 		if (!slot) return;
-		return slot.assignedNodes({ flatten: true }).filter((node) => {
+
+		const results = await this._filterAsync(slot.assignedNodes({ flatten: true }), async node => {
 			if (node.nodeType !== Node.ELEMENT_NODE) return false;
+			await node.updateComplete;
+
 			const role = node.getAttribute('role');
+			if (role !== 'listitem') return false;
+
+			if (this.clearable) node.setAttribute('clearable', 'clearable');
+			node.addEventListener('d2l-tag-list-item-cleared', this._handleItemDeleted.bind(this));
 			node.removeAttribute('data-is-chomped');
-			return (role === 'listitem');
+
+			return true;
 		});
+		return results;
+	}
+
+	_getVisibleEffectiveChildren() {
+		if (!this.shadowRoot) {
+			return [];
+		}
+
+		const showMoreButton = this.shadowRoot.querySelector('.d2l-tag-list-button') || [];
+		const clearButton = this.shadowRoot.querySelector('.d2l-tag-list-clear-button') || [];
+		return this._items.slice(0, this._chompIndex).concat(showMoreButton).concat(clearButton);
+	}
+
+	_handleClearAll(e) {
+		if (!this._items) return;
+
+		this._items.forEach((item) => {
+			item.handleClearItem(e);
+		});
+	}
+
+	_handleItemDeleted(e) {
+		if (!this.clearable) return;
+		if (!e || !e.detail || !e.detail.handleFocus) return;
+
+		const rootTarget = event.composedPath()[0];
+		const children = this._getVisibleEffectiveChildren();
+		const itemIndex = children.indexOf(rootTarget);
+		if (children.length > 1) {
+			if (children[itemIndex - 1]) children[itemIndex - 1].focus();
```

Comment:
I guess to my Q above... what happens when `children.length === 0`?

Reply:
The focus seems to stay on `d2l-tag-list` but there's nothing visible so it is a bit weird. Perhaps it should go to the next focusable element in the document? I'll check with Noah on this.

Reply:
If there was a way to add tags, I might expect the focus to get moved to that, but I am not sure.  Or maybe the consumer listens for a clear event and manages the focus themselves.  Not sure.  🤔

Reply:
There is a future story to add the tag input so agreed that if that existed, then it would make sense to send focus there. I think it would likely make the most sense for the consumer to handle focus in that case. We could leave handling the focus entirely to the consumer, or just note that they would need to handle it for when there are no remaining tag list items.

Reply:
It also seems like a weird configuration where a user would be able to remove tags, but not add them back.